Tests failling on master

```
>           gitchangelog = self.new_release.get('gitchangelog')
E           AttributeError: 'NewRelease' object has no attribute 'get'
```

Full log:
https://travis-ci.org/user-cont/release-bot/jobs/534685599

--- a/README.md
+++ b/README.md
@@ -140,7 +140,7 @@ Here are the `conf.yaml` configuration options:
 | `github_app_cert_path`       | Path to a certificate which Github provides as an auth mechanism for Github apps. | No |
 | `refresh_interval`           | Time in seconds between checks on repository. Default is 180 | No |
 | `clone_url`                  | URL used to clone your Github repository. By default, `https` variant is used. | No |
-| `use_gitchangelog`           | Whether to use gitchanelog to generate change logs. False by default | No |
+| `gitchangelog`               | Whether to use gitchangelog to generate change logs. False by default. | No |
 
 Sample config named [conf.yaml](conf.yaml) can be found in this repository.
 

--- a/release_bot/configuration.py
+++ b/release_bot/configuration.py
@@ -50,6 +50,7 @@ class Configuration:
         self.github_app_cert_path = ''
         self.clone_url = ''
         self.webhook_handler = False
+        self.gitchangelog = False
 
     def set_logging(self,
                     logger_name="release-bot",
@@ -127,7 +128,6 @@ class Configuration:
         # If pypi option is not specified in release-conf.yaml,
         # it defaults to true.
         parsed_conf.setdefault('pypi', True)
-        parsed_conf.setdefault('gitchangelog', False)
 
         parsed_conf = {k: v for (k, v) in parsed_conf.items() if v}
         for item in self.REQUIRED_ITEMS['release-conf']:

--- a/release_bot/git.py
+++ b/release_bot/git.py
@@ -52,6 +52,7 @@ class Git:
         Utilizes [GitChangeLog](https://github.com/vaab/gitchangelog/) to get
         log since latest release according to the template provided.
         :param latest_version: previous version
+        :param gitchangelog: bool, use gitchangelog
         :return: changelog or placeholder
         """
         if gitchangelog:

--- a/release_bot/github.py
+++ b/release_bot/github.py
@@ -420,6 +420,7 @@ class Github:
         Makes the steps to prepare new branch for the release PR,
         like generating changelog and updating version
         :param new_pr: dict with info about the new release
+        :param gitchangelog: bool, use gitchangelog
         :return: True on success, False on fail
         """
         repo = new_pr.repo
@@ -440,9 +441,9 @@ class Github:
             # The bot first checks out the master branch and from master
             # it creates the new branch, checks out to it and then perform the release
             # This makes sure that the new release_pr branch has all the commits
-            # from the master branch for the lastest release.
+            # from the master branch for the latest release.
             repo.checkout('master')
-            changelog = repo.get_log_since_last_release(new_pr['previous_version'], gitchangelog)
+            changelog = repo.get_log_since_last_release(new_pr.previous_version, gitchangelog)
             repo.checkout_new_branch(branch)
             changed = look_for_version_files(repo.repo_path, new_pr.version)
             if insert_in_changelog(f'{repo.repo_path}/CHANGELOG.md',

--- a/release_bot/init_repo.py
+++ b/release_bot/init_repo.py
@@ -57,10 +57,10 @@ class Init:
             'github_token': '<your_github_token>',
             'refresh_interval': '180',
             'github_username': '<your_github_username>',
+            'gitchangelog': False,
         }
         self.release_conf = {
             'trigger_on_issue': True,
-            'gitchangelog': True,
             'author_email': '<your_email>',
             'author_name': '<your_name>',
             'labels': []
@@ -72,7 +72,7 @@ class Init:
         """
         self.create_conf(silent)
         self.append_to_gitignore()
-        if self.release_conf['gitchangelog']:
+        if self.conf['gitchangelog']:
             self.create_template()
             self.create_gitchangelog_rc()
         if silent:
@@ -86,7 +86,6 @@ Please commit all of the changes made to the repo and
 from shell run 'release-bot -c conf.yaml'"""
 
         print(conclude)
-
 
     def create_conf(self, silent):
         """
@@ -120,9 +119,9 @@ from shell run 'release-bot -c conf.yaml'"""
             gitchangelog = input(
                 'Would you like to use gitchangelog to generate next-gen changelogs? (Y/n):'
                 )
-            self.release_conf['gitchangelog'] = bool(
+            self.conf['gitchangelog'] = bool(
                 gitchangelog.lower() == 'y' or gitchangelog == ''
-                )
+            )
 
         self.create_yaml(self.conf, 'conf.yaml')
         self.create_yaml(self.release_conf, 'release-conf.yaml')

--- a/release_bot/releasebot.py
+++ b/release_bot/releasebot.py
@@ -45,8 +45,6 @@ class ReleaseBot:
         self.github = Github(configuration, self.git)
         self.pypi = PyPi(configuration, self.git)
         self.logger = configuration.logger
-        # FIXME: it's cumbersome to work with these dicts - it's unclear how the content changes;
-        #        get rid of them and replace them with individual variables
         self.new_release = NewRelease()
         self.new_pr = NewPR()
 
@@ -202,8 +200,7 @@ class ReleaseBot:
             self.new_pr.repo = self.git
             if not self.new_pr.repo:
                 raise ReleaseException("Couldn't clone repository!")
-            gitchangelog = self.new_release.get('gitchangelog')
-            if self.github.make_release_pr(self.new_pr, gitchangelog):
+            if self.github.make_release_pr(self.new_pr, self.conf.gitchangelog):
                 pr_handler(success=True)
                 return True
         except ReleaseException:

--- a/tests/src/conf_with_gitchangelog.yaml
+++ b/tests/src/conf_with_gitchangelog.yaml
@@ -1,0 +1,6 @@
+repository_name: random_repo
+repository_owner: repo_owner
+github_token: XXXXXXXXXXXXXXXXXXXXXX
+github_username: random_user
+refresh_interval: 60
+gitchangelog: True

--- a/tests/test_load_local_conf.py
+++ b/tests/test_load_local_conf.py
@@ -43,13 +43,18 @@ class TestLoadLocalConf:
         return Path(__file__).parent / "src/conf_with_clone_url.yaml"
 
     @pytest.fixture
+    def conf_with_gitchangelog(self):
+        """Returns a valid configuration with clone_url option"""
+        return Path(__file__).parent / "src/conf_with_gitchangelog.yaml"
+
+    @pytest.fixture
     def sample_conf(self):
         """Return a sample configuration file"""
         return Path(__file__).parent / "src/sample_conf.yaml"
 
     @pytest.fixture
     def non_existing_conf(self):
-        """Return a non-existing configutation file"""
+        """Return a non-existing configuration file"""
         return ""
 
     def test_non_existing_conf(self):
@@ -74,7 +79,7 @@ class TestLoadLocalConf:
         assert configuration.clone_url == 'https://github.com/repo_owner/random_repo.git'
 
     def test_missing_required_items(self, sample_conf):
-        """Tests if missing requied items generate an error"""
+        """Tests if missing required items generate an error"""
         old_value = configuration.REQUIRED_ITEMS['conf']
         configuration.REQUIRED_ITEMS['conf'] = ['test-key']
         configuration.configuration = sample_conf
@@ -83,3 +88,8 @@ class TestLoadLocalConf:
         configuration.REQUIRED_ITEMS['conf'] = old_value
         assert error.type == SystemExit
         assert error.value.code == 1
+
+    def test_conf_with_gitchangelog(self, conf_with_gitchangelog):
+        configuration.configuration = conf_with_gitchangelog
+        configuration.load_configuration()
+        assert configuration.gitchangelog

